plutarch-extra/Plutarch/Extra/Api.hs

```diff
@@ -0,0 +1,90 @@
+module Plutarch.Extra.Api (
+  getContinuingOutputs,
+  findOwnInput,
+  findDatum,
+) where
+
+import GHC.Records (HasField (getField))
+
+import Plutarch.Api.V1 (
+  PAddress,
+  PDatum,
+  PDatumHash,
+  PScriptContext,
+  PScriptPurpose (PSpending),
+  PTuple,
+  PTxInInfo,
+  PTxInfo,
+  PTxOut,
+  PTxOutRef,
+ )
+import Plutarch.Prelude
+
+import Plutarch.Extra.Monad (tmatch)
+
+-- | Find the output txns corresponding to the input being validated.
+getContinuingOutputs :: Term s (PScriptContext :--> PBuiltinList PTxOut)
```

Comment:
If going for the "maximally focused input" route, this function should take the OutRef from `Spending _` and the `outputs` field instead of the whole context.